src/Optimization/hiopAlgFilterIPM.hpp

```diff
@@ -67,11 +67,15 @@
 #include "hiopPDPerturbation.hpp"
 #include "hiopFactAcceptor.hpp"
 
+#ifdef HIOP_USE_AXOM
+#include <axom/sidre/core/DataStore.hpp>
```

Comment:
```suggestion
namespace axom {
namespace sidre {
class DataStore; // forward declaration
}
}
```
`hiopAlgFilterIPM.hpp` only needs a forward declaration, so it's better to avoid including `DataStore.hpp` in a client-facing header file.